jenkins complains: 

```
************* Module ceph_salt.config_shell
ceph_salt/config_shell.py:1246:9: C0303: Trailing whitespace (trailing-whitespace)
ceph_salt/config_shell.py:1247:35: C0303: Trailing whitespace (trailing-whitespace)
ceph_salt/config_shell.py:1239:25: W1202: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
ceph_salt/config_shell.py:1242:31: W1202: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
ceph_salt/config_shell.py:1245:25: W1202: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
ceph_salt/config_shell.py:1250:15: W0703: Catching too general exception Exception (broad-except)
```

--- a/ceph_salt/config_shell.py
+++ b/ceph_salt/config_shell.py
@@ -4,6 +4,7 @@ import itertools
 import logging
 import fnmatch
 import json
+import os
 from pathlib import Path
 import random
 import string
@@ -1227,8 +1228,20 @@ def generate_config_shell_tree(shell):
 class CephSaltConfigShell(configshell.ConfigShell):
     # pylint: disable=anomalous-backslash-in-string
     def __init__(self):
-        super(CephSaltConfigShell, self).__init__(
-            '~/.ceph_salt_config_shell')
+        #
+        # before initializing configshell_fb, check for zero-length prefs.bin
+        configshell_path = os.path.expanduser('~/.ceph_salt_config_shell')
+        prefs_bin_path = '{}/prefs.bin'.format(configshell_path)
+        if os.path.exists(prefs_bin_path):
+            logger.debug('%s exists', prefs_bin_path)
+            prefs_bin_filesize = os.path.getsize(prefs_bin_path)
+            if prefs_bin_filesize == 0:
+                logger.warning('%s has size 0; removing!', prefs_bin_path)
+                os.remove(prefs_bin_path)
+        else:
+            logger.debug('%s does not exist', prefs_bin_path)
+        super(CephSaltConfigShell, self).__init__(configshell_path)
+        #
         # Grammar of the command line
         command = locatedExpr(Word(alphanums + '_'))('command')
         var = QuotedString('"') | QuotedString("'") | Word(alphanums + '?;&*$!#,=_\+/.<>()~@:-%[]')

--- a/ceph_salt/config_shell.py
+++ b/ceph_salt/config_shell.py
@@ -8,6 +8,7 @@ import os
 from pathlib import Path
 import random
 import string
+import sys
 
 
 from pyparsing import alphanums, OneOrMore, Optional, Regex, Suppress, Word, QuotedString
@@ -1240,7 +1241,13 @@ class CephSaltConfigShell(configshell.ConfigShell):
                 os.remove(prefs_bin_path)
         else:
             logger.debug('%s does not exist', prefs_bin_path)
-        super(CephSaltConfigShell, self).__init__(configshell_path)
+        #
+        # initialize configshell_fb
+        try:
+            super(CephSaltConfigShell, self).__init__(configshell_path)
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.exception("configshell_fb failed to initialize")
+            sys.exit("configshell_fb failed to initialize: {}".format(str(exc)))
         #
         # Grammar of the command line
         command = locatedExpr(Word(alphanums + '_'))('command')

